tests/main/hook-permissions/task.yaml

```diff
@@ -0,0 +1,43 @@
+summary: Ensure that post-refresh hook has permissions for its connected slot.
+
+# Note: it is critical for this test to use a slot that requires
+# peer=<plug security tag> expression label for dbus, and not simply
+# label=unconfined on a permanent slot snippet.
+details: |
+    Ensure that post-refresh hook has permissions to query upower-observe slot
+    and enumerate upower devices.
+
+prepare: |
+    # shellcheck source=tests/lib/systems.sh
+    . "$TESTSLIB/systems.sh"
+    if is_core_system; then
+        snap install upower
+    fi
+    if [[ "$SPREAD_SYSTEM" == arch-* ]]; then
+      pacman -S --noconfirm upower
```

Comment:
it should go in pkgdb.sh script. For example in line https://github.com/snapcore/snapd/blob/master/tests/lib/pkgdb.sh#L769 are listed all the packages to be installed in arch linux while the suite is being prepared. The idea is to install all the dependencies during prepare and then run all the tests with the same packages installed. In case for some reason you can't install while the suite is being prepare, so you need to remove the package installed while the test is being restored.

Reply:
Aah, thanks, will do